Core/src/org/sleuthkit/autopsy/modules/hashdatabase/HashDbManager.java

```diff
@@ -786,48 +1031,340 @@ public boolean equals(Object obj) {
             if (getClass() != obj.getClass()) {
                 return false;
             }
-            final HashDb other = (HashDb) obj;
+            final SleuthkitHashSet other = (SleuthkitHashSet) obj;
             if (!Objects.equals(this.hashSetName, other.hashSetName)) {
                 return false;
             }
-            if (this.searchDuringIngest != other.searchDuringIngest) {
+            if (this.knownFilesType != other.knownFilesType) {
+                return false;
+            }
+            return true;
+        }
+    }
+
+    /**
+     * Instances of this class represent hash databases used to classify files
+     * as known or know bad.
+     */
+    class CentralRepoHashSet extends HashDb{
+
+        private static final long serialVersionUID = 1L;
+        private final String hashSetName;
+        private boolean searchDuringIngest;
+        private boolean sendIngestMessages;
+        private final SleuthkitHashSet.KnownFilesType knownFilesType;  
+        private final int referenceSetID;
+        private final String version;
+        private String orgName;
+        private final boolean readOnly;
+        private final PropertyChangeSupport propertyChangeSupport = new PropertyChangeSupport(this);
+
+        @Messages({"HashDbManager.CentralRepoHashDb.orgError=Error loading organization"})
+        private CentralRepoHashSet(String hashSetName, String version, int referenceSetID, 
+                boolean useForIngest, boolean sendHitMessages, SleuthkitHashSet.KnownFilesType knownFilesType, 
+                boolean readOnly)
+                throws TskCoreException{
+            this.hashSetName = hashSetName;
+            this.version = version;
+            this.referenceSetID = referenceSetID;
+            this.searchDuringIngest = useForIngest;
+            this.sendIngestMessages = sendHitMessages;
+            this.knownFilesType = knownFilesType;
+            this.readOnly = readOnly;
+            
+            try{
+                orgName = EamDb.getInstance().getReferenceSetOrganization(referenceSetID).getName();
+            } catch (EamDbException ex){
+                Logger.getLogger(SleuthkitHashSet.class.getName()).log(Level.SEVERE, "Error looking up central repository organization", ex); //NON-NLS
```

Comment:
I think it could help in debugging if we identify the database that experienced the error.